### PR TITLE
Feature/command

### PR DESCRIPTION
**Pull request by IhostVlad**

### CODE REVIEW

File: packages/resolve-command/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "mocha",
+    "testw": "mocha -w"
+  },
+  "author": "",
+  "license": "MIT",
+  "devDependencies": {
+    "babel-preset-es2015": "^6.24.1",
```
Comment (DmitryBogomolov) [inline packages/resolve-command/package.json]: This and the following one are declared in the top level *package.json*.

File: packages/resolve-command/test/index.tests.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            const storedEvents = [];
+            return store.loadEventsByAggregateId('test-id', event => storedEvents.push(event))
+                .then(() => expect(storedEvents)
+                    .to.be.deep.equal([testEvent])
+                );
+        });
+    });
+
+    it('should reject event in case of aggregateName absense', () => {
+        Reflect.deleteProperty(testCommand, '__aggregateName');
```
Comment (DmitryBogomolov) [inline packages/resolve-command/test/index.tests.js]: Why not `delete testCommand.__aggregateName` or `testCommand.__aggregateName = null;`?

File: packages/resolve-command/test/index.tests.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        return execute(testCommand)
+            .then(() => expect(false).to.be.true)
+            .catch(err => expect(err).to.be.equal('Miss __commandName argument'));
+    });
+
+    it('should reject event in case of aggregateId absense', () => {
+        Reflect.deleteProperty(testCommand, '__aggregateId');
+
+        return execute(testCommand)
+            .then(() => expect(false).to.be.true)
```
Comment (DmitryBogomolov) [inline packages/resolve-command/test/index.tests.js]: What is the purpose of this line? (isn't `execute` is rejected in this test?)

File: packages/resolve-command/src/index.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+const INITIAL_STATE_FUNC = '__initialState';
+const BUILD_STATE_FUNC = '__applyEvent';
+
+function verifyCommand(command) {
+    if (!command.__aggregateName) return Promise.reject('Miss __aggregateName argument');
+    if (!command.__aggregateId) return Promise.reject('Miss __aggregateId argument');
+    if (!command.__commandName) return Promise.reject('Miss __commandName argument');
```
Comment (timbset) [inline packages/resolve-command/src/index.js]: `__fieldName argument is required`

File: packages/resolve-es-memory/.babelrc [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (DmitryBogomolov) [inline packages/resolve-es-memory/.babelrc]: There already is *.babelrc* in the top level.

File: packages/resolve-command/src/index.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    return Promise.resolve(command);
+}
+
+function getAggregateState(aggregate, aggregateId, store) {
+    const initialStateFunc = aggregate[INITIAL_STATE_FUNC] || (() => ({}));
+    const buildStateFunc = aggregate[BUILD_STATE_FUNC] || (state => state);
+
+    let aggregateState = initialStateFunc();
+
+    return store.loadEventsByAggregateId(aggregateId, (event) => {
```
Comment (DmitryBogomolov) [inline packages/resolve-command/src/index.js]: If state building function is not defined then there is no need to load events.

File: packages/resolve-command/src/index.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+function saveEvent(event, store) {
+    return store.saveEvent(event).then(() => event);
+}
+
+function publishEvent(event, bus) {
+    bus.emitEvent(event);
+
+    return Promise.resolve(event);
```
Comment (DmitryBogomolov) [inline packages/resolve-command/src/index.js]: Just `return event` (according to the way the function is called).

File: packages/resolve-command/package.json [hunk 8]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "resolve-command",
+  "version": "0.0.1",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
```
Comment (DmitryBogomolov) [inline packages/resolve-command/package.json]: Add *prepublish* script entry.

File: packages/resolve-command/package.json [hunk 9]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "resolve-command",
+  "version": "0.0.1",
+  "description": "",
+  "main": "index.js",
```
Comment (alexey-semikozov) [inline packages/resolve-command/package.json]: Fix main, should be ./dist/index.js

File: packages/resolve-command/src/index.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    if (!command.__aggregateName) return Promise.reject('Miss __aggregateName argument');
+    if (!command.__aggregateId) return Promise.reject('Miss __aggregateId argument');
+    if (!command.__commandName) return Promise.reject('Miss __commandName argument');
+
+    return Promise.resolve(command);
+}
+
+function getAggregateState(aggregate, aggregateId, store) {
+    const initialStateFunc = aggregate[INITIAL_STATE_FUNC] || (() => ({}));
+    const buildStateFunc = aggregate[BUILD_STATE_FUNC] || (state => state);
```
Comment (alexey-semikozov) [inline packages/resolve-command/src/index.js]:
```
aggregate[BUILD_STATE_FUNC] || (state => state);
```
Default buildStateFunc is not check in tests

File: packages/resolve-command/test/index.tests.js (hunk 3 above)
Comment (IhostVlad) [inline packages/resolve-command/test/index.tests.js]: Because in another case (if exception wont be fired) test will be passed

**Pull request merged**